core/src/io/anuke/mindustry/ui/fragments/BlockInventoryFragment.java

```diff
@@ -193,10 +193,14 @@ public void touchUp(InputEvent event, float x, float y, int pointer, KeyCode but
 
     private String round(float f){
         f = (int)f;
-        if(f >= 1000000){
+        if(f >= 10000000){
             return (int)(f / 1000000f) + "[gray]mil[]";
-        }else if(f >= 1000){
+        }else if(f >= 1000000){
+            return (Math.floor(f / 100000) / 10f) + "[gray]mil[]";
+        }else if(f >= 10000){
             return (int)(f / 1000) + "k";
+        }else if(f >= 1000){
+            return (Math.floor(f / 100) / 10) + "k";
```

Comment:
This should be formatted. Even if a potential result here looks like `10.2`, it *may* show up as `10.200003` or something similar when you turn it into a string.